src/transactions/CategoryFilter.js

```diff
@@ -0,0 +1,52 @@
+import React, { useState } from 'react';
+import { PropTypes } from 'prop-types';
+import { useSelector } from 'react-redux';
+
+/**
+ * Displays a select dropdown of categories as options to filter by and
+ * a clear button to reset the dropdown.
+ *
+ * @component
+ */
+function CategoryFilter({ handleFilter }) {
+  const [selected, setSelected] = useState({ selected: 'Filter by Category' });
+  const onSelectChanged = ({ target: { value } }) => {
+    handleFilter({ categoryId: value });
+    setSelected({ selected: value });
+  };
+
+  const onClear = () => {
+    handleFilter({ categoryId: null });
+    setSelected({ selected: 'Filter by Category' });
+  };
+
+  const categoryList = useSelector((state) => state.categories.entities);
+
+  return (
+    <div className="row">
+      <div className="col col-3 mr-2">
+        <select className="form-select" value={selected.selected} name="categoryFilter" onChange={onSelectChanged}>
+          <option value="Filter by Category">Filter by Category</option>
```

Comment:
Instead of the value "Filter by Category" which is easy to mistype and could lead to bugs, can we set the value to an empty string? That's a common way to set the value of the none option.